ingestion/src/metadata/profiler/processor/sampler/sqlalchemy/sampler.py

```diff
@@ -63,26 +63,20 @@ class SQASampler(SamplerInterface):
     run the query in the whole table.
     """
 
-    @partition_filter_handler(build_sample=True)
-    def get_sample_query(self) -> Query:
-        """get query for sample data"""
-        if self.profile_sample_type == ProfileSampleType.PERCENTAGE:
-            rnd = (
-                self.client.query(
-                    self.table,
-                    (ModuloFn(RandomNumFn(), 100)).label(RANDOM_LABEL),
-                )
-                .suffix_with(
-                    f"SAMPLE BERNOULLI ({self.profile_sample or 100})",
-                    dialect=Dialects.Snowflake,
-                )
-                .cte(f"{self.table.__tablename__}_rnd")
+    def get_sample_percentage_cte(self):
+        return (
+            self.client.query(
+                self.table,
+                (ModuloFn(RandomNumFn(), 100)).label(RANDOM_LABEL),
             )
-            session_query = self.client.query(rnd)
-
-            return session_query.where(rnd.c.random <= self.profile_sample).cte(
-                f"{self.table.__tablename__}_sample"
+            .suffix_with(
+                f"SAMPLE BERNOULLI ({self.profile_sample or 100})",
+                dialect=Dialects.Snowflake,
             )
+            .cte(f"{self.table.__tablename__}_rnd")
+        )
+
+    def get_sample_rows_cte(self):
```

Comment:
We don't need both function you can have one return the main difference and pass the random function as the parameter, right?